### PR TITLE
Fix invoke method interface. 

### PR DESCRIPTION
**Pull request by KSchopmeyer**
I misinterpreted the Params vs params options in invoke method and did not incorporate both Params and params into fake  InvokeMethods cleanly.  I fact, we really do not have an example that shows the various options for calling InvokeMethod  to make it easy for the client writer but that is a different question.

This cleans up the InvokeMethod and callback interface and adds more tests.

### CODE REVIEW

File: pywbem_mock/_wbemconnection_mock.py [hunk 1]
```diff
@@ -91,8 +91,8 @@ def _display(dest, text):
         f.close()
 
 
-def method_callback_interface(conn, objectname, methodname, params):
-    # pylint: disable=unused-argument
+def method_callback_interface(conn, objectname, methodname, Params, **params):
+    # pylint: disable=unused-argument, invalid-name
```
Comment (andy-maier) [inline pywbem_mock/_wbemconnection_mock.py]: Having both a Params argument and kwargs on the callback interface is really a historical relic on InvokeMethod that we did not simplify for compatibility reasons.

Technically, we could have just a single param argument on the callback interface, that is an iterable of CIMParameter objects. Please think about whether it is important to represent the exact interface of InvokeMethod on the callback (which requires the user to first merge the two lists) vs. simplifying it.

Comment (KSchopmeyer) [reply pywbem_mock/_wbemconnection_mock.py]: Let’s go for the single interface in the callback method. I assume that it should be the iterable if CIMParameter and we will merge the two inputs. However. Lets get his fix in and also the CIM paramount fix and then I will modify.

Comment (andy-maier) [reply pywbem_mock/_wbemconnection_mock.py]: ok. merging it.

Comment (KSchopmeyer) [reply pywbem_mock/_wbemconnection_mock.py]: Tks.

**Pull request merged**